src/process_utility.c

```diff
@@ -783,6 +776,14 @@ add_chunk_to_vacuum(Hypertable *ht, Oid chunk_relid, void *arg)
 	chunk_vacuum_rel =
 		makeVacuumRelation(chunk_range_var, chunk_relid, ctx->ht_vacuum_rel->va_cols);
 	ctx->chunk_rels = lappend(ctx->chunk_rels, chunk_vacuum_rel);
+
+	/* If we have a compressed chunk, make sure to analyze it as well */
+	if (chunk->fd.compressed_chunk_id != INVALID_CHUNK_ID)
+	{
+		Chunk *comp_chunk = ts_chunk_get_by_id(chunk->fd.compressed_chunk_id, true);
```

Comment:
IMHO we should set the `fail_if_not_found` to `false` for those `ts_chunk_get_by_id` calls and do a proper check after to skip it if necessary... think about an hypertable with lot of chunks and concurrent jobs doing compression/decompression and/or retention... it means that if for some reason after build the chunk list some of then be removed in other session we'll fail and left other chunks behind without ANALYZE... is this our expected behavior??

Reply:
You are right, I'll update to continue if a compressed chunk is not found.